### PR TITLE
Add config to support viewfs in Gluten.

### PR DESCRIPTION
**Pull request by JkSelf**
## What changes were proposed in this pull request?

To enable support for ViewFS in Gluten, we need to call FileSystem.get(viewPath.toUri, serializableHadoopConf.value) to convert the ViewFS path to an HDFS path. Prior to this PR, we passed serializableHadoopConf in getSplitInfosFromPartitions for the conversion. This PR updates the path values after obtaining the LocalFilesNode, thereby reducing the number of times serializableHadoopConf needs to be passed. With this PR, Jenkins performance can improve by about 5%.

## How was this patch tested?

Jenkins test

### CODE REVIEW

**Comment by FelixYBW** [comment]
@JkSelf can you add some desc why the PR revert view fs?

File: shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala [hunk 1]
```diff
@@ -2170,4 +2172,11 @@ object GlutenConfig {
           "Otherwise, do nothing.")
       .booleanConf
       .createWithDefault(false)
+
+  val HDFS_VIEWFS_ENABLED =
+    buildStaticConf("spark.gluten.storage.hdfs.viewfs.enabled")
```
Comment (PHILO-HE) [inline shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala:2177]: Some of these words can be combined with no period mark, which may look better.
-> `spark.gluten.storage.hdfsViewfs.enabled`

File: shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala [hunk 2]
```diff
@@ -2170,4 +2172,11 @@ object GlutenConfig {
...
+      .internal()
+      .doc("If enabled, gluten will convert the viewfs path to hdfs path in scala size")
```
Comment (PHILO-HE) [inline shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala:2179]: typo, size -> side?

**Review by PHILO-HE — APPROVED** [review]
Just two minor comments. Thanks!

**Comment by JkSelf** [comment]
@PHILO-HE Thanks for your review. Resolved all your comments. Can you help to approve again? Thanks.

**Review by PHILO-HE — APPROVED** [review]
Thanks!

**Pull request merged**